Sources/Submissions/SubmissionType.swift

```diff
@@ -127,25 +139,28 @@ extension SubmissionType {
 }
 
 extension BasicValidationError {
-
      /// The default error to throw when a value is empty when that is not valid.
-     public static var onNil: BasicValidationError {
+     public static var onEmpty: BasicValidationError {
         return .init("Value may not be empty")
     }
 }
 
 extension Future where T: SubmissionType {
-    
     /// Convenience for calling `validate` on submissions produced by this `Future`.
     ///
     /// - Parameters:
+    ///   - submittable: An optional existing related submittable for reference when validating.
     ///   - context: The context (update/create) to respect when validating
-    ///   - container: The container with the event loop to validate on and the field cache to store
+    ///   - req: The `Request` with the event loop to validate on and the field cache to store
     ///     any validation errors.
     /// - Returns: A `Future` of the `SubmissionType` value.
-    public func validate(inContext context: ValidationContext, on container: Container) -> Future<T> {
+    public func validate(
```

Comment:
I forget, did I have the order different here? Now it reads as if you're validating the submittable, which it not the case. The submittable is only for reference.

Reply:
I agree, updated to `with submittable` and moved it to be the second argument.